File: Python/two_sum.py
# Runtime: Embarrassing
# Memory: 18MB (51.32%)

from typing import List

def twoSum(nums: List[int], target: int) -> List[int]:
    for i in nums:
        for j in nums:
            if nums.index(i) != nums.index(j) and i + j == target:
                return [nums.index(i), nums.index(j)]

def twoSumTwo(nums: List[int], target: int) -> List[int]:
    length = len(nums)
    for i in range(0, length, 1):
        for j in range(0, length, 1):
            if i != j and nums[i] + nums[j] == target:
                return [i, j]

print(twoSumTwo([2,7,11,15], 9))
print(twoSumTwo([3,2,4], 6))
print(twoSumTwo([3,3], 6))

File: C#/two_sum.cs
public class Solution
{
    public static int[] TwoSum(int[] nums, int target)
    {
        for(int i = 0; i < nums.Length; i++)
        {
            for(int j = 1; j < nums.Length; j++)
            {
                if (nums[i] + nums[j] == target && i != j)
                {
                    return [i, j];
                }
            }
        };
        return [0];
    }

    static public void Main(String[] args)
    {
        int[] soln = TwoSum([2, 7, 11, 15], 9);
        Console.Write("[ ");
        for(int i = 0; i < soln.Length; i++)
        {
            Console.Write(soln[i] + " ");
        }
        Console.Write("]");
    }
}